src/components/DisplayArticleById.jsx

```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import SingleArticle from "./SingleArticle";
+
+const DisplayArticleById = () => {
+  const { article_id } = useParams();
+  const [currentArticle, setCurrentArticle] = useState({});
+  const [isLoading, setIsLoading] = useState("");
```

Comment:
You have a loading state here that isn't being utilised, Loading state for your page would be a great feature for an improved user experience